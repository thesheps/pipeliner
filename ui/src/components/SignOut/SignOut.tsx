import React from "react";
import Button from "@material-ui/core/Button";
import { connect } from "react-redux";

import { signOutUserThunk } from "../../store/user/thunks";

export interface SignOutProps {
  signOutUser: () => void;
}

export const SignOut = ({ signOutUser }: SignOutProps) => (
  <Button
    data-testid="signOut-button"
    color="inherit"
    onClick={() => signOutUser()}
  >
    Sign Out
  </Button>
);

export const SignOutContainer = connect(null, {
  signOutUser: signOutUserThunk,
})(SignOut);
