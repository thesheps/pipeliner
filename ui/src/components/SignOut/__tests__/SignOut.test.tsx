import React from "react";
import { fireEvent, render } from "@testing-library/react";
import {
  toBeDisabled,
  toBeVisible,
  toHaveStyle,
} from "@testing-library/jest-dom/matchers";

import { SignOut } from "../SignOut";

expect.extend({ toBeDisabled, toBeVisible, toHaveStyle });

describe("SignOut", () => {
  it("renders correctly", () => {
    const { container } = render(<SignOut signOutUser={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });

  it("calls signOutUser on signOut click", () => {
    const signOut = jest.fn();
    const { getByText } = render(<SignOut signOutUser={signOut} />);

    fireEvent.click(getByText("Sign Out"));

    expect(signOut).toHaveBeenCalled();
  });
});
