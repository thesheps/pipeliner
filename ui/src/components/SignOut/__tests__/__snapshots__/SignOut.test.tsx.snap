// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut renders correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    data-testid="signOut-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Sign Out
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
